python/lsst/meas/astrom/matchOptimisticB/matchOptimisticBContinued.py

```diff
@@ -286,6 +287,12 @@ def matchObjectsToSources(self, refCat, sourceCat, wcs, refFluxField, maxMatchDi
             self.log.info("filterStars purged %d reference stars, leaving %d stars" %
                           (preNumObj - numRefObj, numRefObj))
 
+        if toleranceStruct is None:
+            toleranceStruct = pipeBase.Struct(
+                maxMatchDist=None,
+            )
```

Comment:
Here is an excellent example of where a custom class helps. You can replace that with:
```
if tolerance is None:
    tolerance = MatchTolerance()
```
and you get a fully populated instance!